### ISSUE TITLE
csc.exe exited with code -2146232797 (Local functions)

### ISSUE THREAD

**Issue by KonstantinRyazantsev** [dotnet/roslyn#23167]
**Version Used**: VS 2017 version 15.3.1

**Steps to Reproduce**:

1. 
```
private static Type GetEnumerableElementType(Type collectionType)
{
    if (collectionType.IsGenericType)
    {
        Type GetEnumerableElementType(Type type)
        {
            return collectionType.GetGenericTypeDefinition() == typeof(IEnumerable<>)
                ? collectionType.GetGenericArguments().First()
                : null;
        }

        Type GetImplementorOfEnumerableElementType(Type type)
        {
            return type
                .GetGenericTypeDefinition()
                .GetInterfaces()
                .Select(GetEnumerableElementType)
                .FirstOrDefault(t => t != null);
        }

        var elementType = GetEnumerableElementType(collectionType) ??
                            GetImplementorOfEnumerableElementType(collectionType);

        return elementType;
    }

    return null;
}
```
2. Build

**Expected Behavior**:

**Actual Behavior**:

Code: MSB6006
Description: "csc.exe" exited with code -2146232797
File: Microsoft.CSharp.Core.targets
Line: 84

**Comment by jcouv** [comment]
I'm able to build the code above in 15.5 preview 4.

@KonstantinRyazantsev Would you mind trying with a preview of 15.5 as well? A number of local functions issues were fixed in 15.5.

FYI @agocke

**Comment by framgeld** [comment]
15.5 Preview 4 corrects the C# compiler issues when using local functions. A workaround of adding the Microsoft.Net.Compilers v2.2.0 NuGet package did not work in my case (although it worked in other cases for us).

Unfortunately, unless 15.5 fixes https://github.com/dotnet/roslyn/issues/22776 also (which is currently slated for a fix in 15.7!), we are stuck between a rock and a hard place. We can either: a) not compile in v15.4 or b) have the v15.5 IDE riddled with invalid Roslyn errors.

VS2017 has quickly turned from one of the best releases of Visual Studio in recent memory, into a bit of a mess due to the updates.

Is there any chance of you guys creating a pre-release NuGet package of Microsoft.Net.Compilers with the fixes to csc.exe so that I can continue to use 15.4? Would going through the official Microsoft support channels guarantee this outcome?

**Comment by jcouv** [comment]
@framgeld With the incremental release approach, we rarely go back and patch older releases. Only a few very severe issues are serviced that way. In most cases, the recommended approach is to pick up the new version of Roslyn with the fix.
Thanks

 #22776 was fixed in 15.5.
I'll go ahead and close the issue, since was fixed in 15.5.
